### PR TITLE
add QA/QC module and instability checker function

### PR DESCRIPTION
**Pull request by bolliger32**
See docstring for info on how the function works. This will definitely be useful for the Climate Impact Lab work, so if you don't think this is a good general function, we're happy to just keep it on our fork. Otherwise, feel free to merge! I'm imagining other potential ways to do QA/QC that could be added to this module in the future. Also happy to rearrange the structure/move to another module if you think there's a better way to structure it.

### CODE REVIEW

File: src/python/geoclaw/surge/quality.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+            # adjust for sl_init
+            all_vals = all_vals - sl_init
+
+            if all_vals.shape[0]>0:
+                if abs(np.nanmean(all_vals)) > mean_tol_abs:
+                    raise InstabilityError("Model possibly unstable due to large magnitude deep "
+                                           "ocean surge at end of run ({:.1f} cm in region {})".format(
+                                           np.nanmean(all_vals)*100, r))
+
+    return None
```
Comment (mandli) [inline src/python/geoclaw/surge/quality.py]: Probably do not need this `return` here unless I missed somewhere that it is also called.  I also might suggest naming the function `quality_check` or something other than stability.

**Comment by mandli** [comment]
I think this could definitely be useful and maybe even beyond the storm surge case but I think it would take more time to figure out how to generalize it.

File: src/python/geoclaw/surge/quality.py (hunk 1 above)
Comment (bolliger32) [reply src/python/geoclaw/surge/quality.py]: cool. Changed name to `quality_check` and removed the extraneous `return`. And agreed on the generality - I don't think I know enough about the structure of non-geoclaw use cases of clawpack to make this more general, so maybe we keep it within the geoclaw submodule for now and think about how to make something like this more general.

Comment (mandli) [reply src/python/geoclaw/surge/quality.py]: The more general applications would still be GeoClaw based but they still would be slightly different depending on if it was for tsunamis, debris-flows, dam-breaks, etc.

**Pull request merged**